topdown/http.go

```diff
@@ -185,15 +185,41 @@ func executeHTTPRequest(bctx BuiltinContext, obj ast.Object) (ast.Value, error)
 		if err != nil {
 			return nil, err
 		}
+
 		key = key.(string)
 
+		strVal, isString := func() (string, bool) {
+			if s, ok := obj.Get(val).Value.(ast.String); ok {
+				return string(s), true
+			}
+			return "", false
+		}()
```

Comment:
Does this need to be wrapped in a function?

I'm also wondering if we can maybe re-use some of the normal built-in type checking/definition code for this. I'm a little bit worried about how maintainable the current stuff is (even without this new string check) as we keep adding more and more options.

Reply:
To clarify, not a blocking issue. My comment about using the other code is just thinking out loud for future cleanup/iteration we can do in here.

Reply:
>Does this need to be wrapped in a function?

Yeh, because if we have a `ast.String` then we need another temp variable or to cast at all the callsites. Both of those options felt more awkward. 

> I'm also wondering if we can maybe re-use some of the normal built-in type checking/definition code for this.

Is the `builtin.*Operand` API what you had in mind?  That looks like a solid improvement to me.

Reply:
I think it can be refactored to something like:

```go
var strVal string

if s, ok := obj.Get(val).Value.(ast.String); ok {
	strVal = string(s)
} else {
	// Most parameters are strings, so consolidate the type checking.
	switch key {
	case "method",
		"url",
		"raw_body",
		"tls_ca_cert",
		"tls_ca_cert_file",
		"tls_ca_cert_env_variable",
		"tls_client_cert",
		"tls_client_cert_file",
		"tls_client_cert_env_variable",
		"tls_client_key",
		"tls_client_key_file",
		"tls_client_key_env_variable",
		"tls_server_name":
		return nil, fmt.Errorf("%q must be a string", key)
	}
}
```

and we avoid any of the function stuff

> Is the builtin.*Operand API what you had in mind? That looks like a solid improvement to me.

Thats part of it, yea. The other part is like the type definitions we use for the builtins themselves. We have a whole set of functionality to define the type of things in Rego and do type checking, I'm wondering how much we can leverage for this (or if we need to.. its a big hammer).

Reply:
That refactor is nice; I applied it and pushed an update.